src/contracts.rs

```diff
@@ -0,0 +1,88 @@
+#![cfg_attr(feature = "dev", feature(plugin))]
+#![cfg_attr(feature = "dev", plugin(clippy))]
+
+#![deny(clippy, clippy_pedantic)]
+#![allow(missing_docs_in_private_items, unknown_lints)]
+
+extern crate serde;
+extern crate serde_json;
+extern crate glob;
+extern crate futures;
+
+use self::glob::glob;
+use self::serde_json::Value;
+pub use keystore::Address;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+/// Contracts Service
+pub struct Contracts {
+    dir: PathBuf,
+}
+
+/// Contract Service Errors
+#[derive(Debug, Clone)]
+pub enum ContractError {
+    /// IO Error
+    IO,
+    /// Invalid Contract
+    InvalidContract,
+}
+
+impl Contracts {
+    /// Initialize new contracts service for a dir
+    pub fn new(dir: PathBuf) -> Contracts {
+        Contracts { dir: dir }
+    }
+
+    fn read_json(path: &Path) -> Result<Value, ContractError> {
+        match File::open(path) {
+            Ok(f) => serde_json::from_reader(f).or(Err(ContractError::IO)),
+            Err(_) => Err(ContractError::IO),
+        }
+    }
+
+    /// List all available contracts
+    pub fn list(&self) -> Vec<Value> {
+        let files = glob(&format!("{}/*.json", &self.dir.to_str().unwrap())).unwrap();
+        files.filter(|x| x.is_ok())
+            .map(|x| Contracts::read_json(x.unwrap().as_path()))
+            .filter(|x| x.is_ok())
+            .map(|x| x.unwrap())
```

Comment:
You can try https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map instead.